cmd/auth.go

```diff
@@ -60,7 +63,14 @@ func login(cmd *cobra.Command, args []string, astroClient astro.Client, coreClie
 	cmd.SilenceUsage = true
 
 	if len(args) == 1 {
+		// check if user provided a valid cloud domain
 		if !context.IsCloudDomain(args[0]) {
+			// get the domain from context as an extra check
+			ctx, _ := context.GetCurrentContext()
+			if context.IsCloudDomain(ctx.Domain) {
+				// return error if context domain is a valid cloud domain
+				return fmt.Errorf("%w: %s. %s", errInvalidDomain, args[0], "You are attempting to login into Astro.\n Are you trying to authenticate to Astronomer Software? If so, please change your current context with 'astro context switch'")
+			}
```

Comment:
why can't we ask for user input at this point in time, a yes/no question? because this error would be hit also for a valid case when a user is logged into Astro and is then trying to log in to a valid Software domain (a valid scenario for almost all migrating customers). This means two extra steps for users which could be avoided I feel.

```
Error: invalid domain: fail.astronomer.io. You are attempting to login into Astro.
 Are you trying to authenticate to Astronomer Software? (y/n)
```

If a user says `yes` at that point in time, then we move on to call `softwareLogin(args[0], oAuth, "", "", houstonVersion, houstonClient, out)` else return out of the function

Reply:
I like the idea. Thanks!